src/components/distributions/editor/TextForm/TextInput.js

```diff
@@ -73,7 +77,7 @@ export default class TextInput extends Component{
     value: PropTypes.string,
   }
 
-  componentWillUnmount() { this._handleBlur() }
+  componentWillUnmount() { this._handleBlur(this.props.value) }
```

Comment:
Is this needed? I would have assumed that if the props.value is the same as what is in the textform (and thus why you call this.props.value here instead of this.state.value), we could obtain the value from above.